src/text/parsers/timestamp.rs

```diff
@@ -270,13 +269,21 @@ fn timezone_offset(input: &str) -> IResult<&str, Option<i32>> {
 }
 
 /// Recognizes the hours and minutes portions of a timezone offset. (`12` and `35` in `12:35`)
-fn timezone_offset_hours_minutes(input: &str) -> IResult<&str, (&str, &str)> {
-    separated_pair(
-        // The parser does not restrict the range of hours/minutes allowed in the offset.
+fn timezone_offset_hours_minutes(input: &str) -> IResult<&str, (i32, i32)> {
+    let (remaining_input, (hours, minutes)) = separated_pair(
         recognize(pair(digit, digit)),
         tag(":"),
         recognize(pair(digit, digit)),
-    )(input)
+    )(input)?;
+    let hour_int = trim_zeros_expect_i32(hours, "offset hours");
+    let minute_int = trim_zeros_expect_i32(minutes, "offset minutes");
+    if hour_int >= 24 {
+        return Err(Err::Failure(make_error(hours, ErrorKind::TooLarge)));
+    }
+    if minute_int >= 60 {
+        return Err(Err::Failure(make_error(minutes, ErrorKind::TooLarge)));
+    }
```

Comment:
🗺️ Previously we only checked that the number of minutes represented by the `offset` we found (`hours * 60 + minutes`) was reasonable. This check occurred in the `Timestamp` constructor. Now we are also checking that the values of the individual `hours` and `minutes` substrings make sense.